presto-main/src/main/java/com/facebook/presto/sql/planner/SymbolReferenceExtractor.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.sql.planner;
+
+import com.facebook.presto.sql.tree.DefaultTraversalVisitor;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.SymbolReference;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SymbolReferenceExtractor
+{
+    private SymbolReferenceExtractor() {}
+
+    public static List<SymbolReference> getSymbolReferences(Expression expression)
+    {
+        SymbolReferenceExtractingVisitor symbolReferenceExtractingVisitor = new SymbolReferenceExtractingVisitor();
+        symbolReferenceExtractingVisitor.process(expression, null);
+        return symbolReferenceExtractingVisitor.getSymbolReferences();
+    }
+
+    private static class SymbolReferenceExtractingVisitor
+            extends DefaultTraversalVisitor<Void, Void>
```

Comment:
> I meant symbols captured by lambda body.

Yes, you need to distinguish symbol references which are lambda argument references from symbol references, which are captures.